pkg/syncer/metadatasyncer.go

```diff
@@ -986,7 +986,10 @@ func csiPVUpdated(ctx context.Context, newPv *v1.PersistentVolume, oldPv *v1.Per
 
 	// Dynamically provisioned PVs have a volume attribute called 'storage.kubernetes.io/csiProvisionerIdentity'
 	// in their CSI spec, which is set by external-provisioner.
-	_, dynamic := newPv.Spec.CSI.VolumeAttributes[attribCSIProvisionerID]
+	var dynamic bool
+	if newPv.Spec.CSI != nil {
+		_, dynamic = newPv.Spec.CSI.VolumeAttributes[attribCSIProvisionerID]
```

Comment:
variable `dynamic` will remain false for in-tree VCP volume and will create ambiguity.
Can you change the name of variable dynamic to `isdynamicCSIPV`?

and also add a check for this condition to ensure the code block is for statically created CSI volume and not for VCP volume?

```golang
if oldPv.Status.Phase == v1.VolumePending && newPv.Status.Phase == v1.VolumeAvailable && !isdynamicCSIPV && newPv.Spec.CSI != nil
```

Reply:
Done